Sources/MapboxDirections/Credentials.swift

```diff
@@ -63,6 +63,17 @@ public struct Credentials: Equatable {
             self.host = URL(string: "https://api.mapbox.com")!
         }
     }
+    
+    public init(requestURL url: URL) {
+        var components = URLComponents(url: url, resolvingAgainstBaseURL: false)
+        let accessToken = components?
+            .queryItems?
+            .first { $0.name == "access_token" }?
+            .value
+        components?.path = "/"
+        components?.queryItems = nil
+        self.init(accessToken: accessToken, host: components?.url)
```

Comment:
Should we return `nil` in case there was an issue while parsing the URL? As at the moment we might implicitly return an object with default values

Reply:
The use-case this method is designed for is fine to subtly fallback to a default. I didn't think this method would be usable or visible to wide auditory, but indeed this case should be at least documented.